internal/component/database_observability/mysql/collector/query_sample.go

```diff
@@ -300,3 +298,75 @@ func (c *QuerySample) fetchQuerySamples(ctx context.Context) error {
 
 	return nil
 }
+
+func (c *QuerySample) calculateWallTime(serverStartTime, timer float64) uint64 {
+	// timer indicates event timing since server startup.
+	// The timer value is in picoseconds with a column type of bigint unsigned. This value can overflow after about ~213 days.
+	// We need to account for this overflow when calculating the timestamp.
+
+	// Knowing the number of overflows that occurred, we can calculate how much overflow time to compensate.
+	previousOverflows := calculateNumberOfOverflows(c.lastUptime)
+	overflowTime := float64(previousOverflows) * picosecondsOverflowInSeconds
+	// We then add this overflow compensation to the server start time, and also add the timer value (remember this is counted from server start).
+	// The resulting value is the timestamp in seconds at which an event happened.
+	timerSeconds := picosecondsToSeconds(timer)
```

Comment:
Why do we convert to seconds here? Aren't we losing precision on this conversion?

Reply:
We are good --using floats so fractional seconds are used.